packages/eslint-config-kibana/.eslintrc.js

```diff
@@ -118,8 +118,9 @@ module.exports = {
     'react/jsx-first-prop-new-line': ['error', 'multiline-multiprop'],
     'react/jsx-equals-spacing': ['error', 'never'],
     'react/jsx-indent': ['error', 2],
-    'react/no-did-mount-set-state': 'error',
-    'react/no-did-update-set-state': 'error',
+    // Off as of React 16.3, see https://github.com/yannickcr/eslint-plugin-react/issues/1754
```

Comment:
I added that react version to the .eslintrc config in Kibana, since the `eslint-config-kibana` actually doesn't have any knowledge about the React version Kibana is using.